bitnami/openldap/2.6/debian-12/rootfs/opt/bitnami/scripts/openldap/run.sh

```diff
@@ -21,14 +21,60 @@ command="$(command -v slapd)"
 # https://github.com/docker/docker/issues/8231
 ulimit -n "$LDAP_ULIMIT_NOFILES"
 
-flags=("-h" "ldap://:${LDAP_PORT_NUMBER}/ ldapi:///")
+declare -a flags
+declare -A flags_map
+
+# Drop privileges if we start as root
+am_i_root && flags_map["-u"]="${LDAP_DAEMON_USER}"
+
+# Set config dir
+flags_map["-F"]="${LDAP_CONF_DIR}/slapd.d"
+
+# Enable debug with desired level
+flags_map["-d"]="${LDAP_LOGLEVEL}"
+
+# The LDAP IPC is always on
+flags_map["-h"]+="${flags_map["-h"]:+" "}ldapi:///"
+
+# Add LDAP URI
+# Since 'proxied LDAP' default port number is same as 'LDAP',
+# enable LDAP URI when one of the following conditions are met:
+# * proxy protocol capability is disabled
+# * proxy protocol capability is enabled and proxy protocol port differ
+if ! is_boolean_yes "${LDAP_ENABLE_PROXYPROTO}" \
+    || [[ "${LDAP_PORT_NUMBER}" -ne "${LDAP_PROXYPROTO_PORT_NUMBER}" ]]
+then
+  flags_map["-h"]+="${flags_map["-h"]:+" "}ldap://:${LDAP_PORT_NUMBER}/"
```

Comment:
As per above comment, we are sure `flags_map["-h"]` will have a value. What if we just do this?

```suggestion
  flags_map["-h"]+=" ldap://:${LDAP_PORT_NUMBER}/"
```

Same suggestion in the concatenations below.

Reply:
Surely for now `flags_map["-h"]` have value, but in case:
- incorporate additional logic (e.g: enabling ldapi:// by condition)
- rearrange "blocks"
- new protocol emerge (e.g: ldap 9.1 now support uldap:// -- LDAP over UDP :upside_down_face: )
all that required -- refactor/move/add "logic block" without deep analyze neighborhood "blocks".

If you think this is overkill, i'm no worries to accept proposed suggestions.